### DISCUSSION TITLE
[Enhancement] End-To-End encrypt everything that can be encrypted

### DISCUSSION THREAD

**Discussion by Myzel394** [simple-login/app#909] [Feature Requests]
Took a quick look at the code and it seems like fields like "Alias Note" are not encrypted.

I think it would offer a lot more privacy when these fields could be encrypted with the public key the user provides on a Premium Plan.
I'm not sure what other fields might be possible to encrypt (because fields such as the email alias should not be encrypted), maybe a member or contributor has more ideas on that.

This E2EE is very beneficial in the case that SimpleLogin gets hacked. At least the thoughts on the aliases wouldn't be breached in such a case.

**Comment by mardom1** [comment]
I would even go a step further and encrypt everything, especially the email log, aliases and mailboxes.

My first thought would be to store a hash of the alias, and the target mailbox(es) encrypted with the plaintext alias (or a key derived from it).

When forwarding, the alias is available in plaintext (mail to). The database entry is fetched by its hash, and the mailboxes decrypted. After being delivered, the plaintext alias is forgotten and the entry cannot be decrypted anymore.

Of course, there is much more to it than that, i.e., with alias auto-creation, and to allow showing it in the user interface. But I personally think this should be a top priority, especially for a privacy oriented service like SimpleLogin. It also significantly reduces the benefit of using Proton Mail, when all correspondence (without mail body of course) is stored in plaintext on SimpleLogin. I paid for premium, but I am not sure I want to use it anymore after looking into the code. Or did I overlook the encryption?

**Reply by mardom1** [reply]
Ok, apparently the Postgres itself encrypts the data at rest. Still, I think end-to-end encryption should be the goal here.

**Reply by Myzel394** [reply]
@mardom1 if you're curious. I created a new email alias service from scratch about a year ago after I opened this issue. It's not really maintained at the moment, but maybe someone wants to pick it up :D Available here: https://github.com/Myzel394/kleckrelay-server
